docker_test.go

```diff
@@ -46,6 +50,65 @@ func init() {
 	}
 }
 
+func TestWithinContainerStage1(t *testing.T) {
+	// TODO: remove this skip check when context rework is merged.
+	if providerType != ProviderDocker {
+		t.Skip("This is a docker-specific test.")
+	}
+	dir, err := os.Getwd()
+	require.NoError(t, err)
+	req := ContainerRequest{
+		Image:      golangImage,
+		WorkingDir: dir,
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.Mounts = append(hc.Mounts, mount.Mount{
+				Type:   "bind",
+				Source: dir,
+				Target: dir,
+			}, mount.Mount{
+				Type:   "bind",
+				Source: "/var/run",
+				Target: "/var/run",
+			})
+		},
+		Entrypoint: []string{"sleep", "100"},
+	}
+	container, err := GenericContainer(context.Background(), GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+	// The timeout is necessary because when TestWithinContainersStage2 fails, it hangs forever instead of erroring
+	exitCode, outputReader, err := container.Exec(context.Background(), []string{"go", "test", "-v", "-run", "^TestWithinContainerStage2$", "-timeout", "20s", "-stage2"}, exec.Multiplexed())
+	require.NoError(t, err)
+
+	outputBytes, err := io.ReadAll(outputReader)
+	require.NoError(t, err, "failed to read output")
+	t.Logf("%s", outputBytes)
```

Comment:
question: looks like this was just debug?

Reply:
Without this, you can't see the output of stage 2, so you don't really know if it's working. I'll remove the extra print in case of error though.

Reply:
Ahh, got you because it's basically a sub test